### PR TITLE
Add code for v4 platform inference and platform inference tests

### PR DESCRIPTION
**Pull request by jkgoodrich**

### CODE REVIEW

File: gnomad_qc/v4/sample_qc/platform_inference.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+                test_scores_path = get_checkpoint_path(
+                    f"test_platform_scores.{args.calling_interval_name}.pad{args.calling_interval_padding}"
+                )
+                if file_exists(test_scores_path):
+                    scores_ht = hl.read_table(test_scores_path)
+                else:
+                    raise FileNotFoundError(
+                        f"The test platform PCA Table does not exist for calling interval "
+                        f"{args.calling_interval_name} and interval padding {args.calling_interval_padding}. "
+                        f"Please run --compute_coverage and -- run_platform_pca with the --test argument and needed "
```
Comment (klaricch) [inline gnomad_qc/v4/sample_qc/platform_inference.py:161]:
```suggestion
                        f"Please run --compute_coverage and --run_platform_pca with the --test argument and needed "
```

**Review by klaricch — CHANGES_REQUESTED** [review]
looks good, just a few small things

File: gnomad_qc/v4/resources/basics.py [hunk 2]
```diff
@@ -216,3 +217,24 @@ def add_meta(
...
+    """
+    if interval_name not in {"ukb", "broad", "intersection"}:
+        ValueError("Calling interval name must be one of: 'ukb', 'broad', or 'intersection'!")
+    if calling_interval_padding not in {0, 50}:
+        ValueError("calling interval padding must be one of: 0 or 50 (bp)!")
+    if interval_name == "ukb":
+        return TableResource(f"gs://gnomad/resources/intervals/xgen_plus_spikein.Homo_sapiens_assembly38.targets.pad{calling_interval_padding}.interval_list.ht")
+    elif interval_name == "broad":
+        return TableResource(f"gs://gnomad/resources/intervals/hg38_v0_exome_calling_regions.v1.pad{calling_interval_padding}.interval_list.ht")
+    elif interval_name == "intersection":
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v4/resources/basics.py:238]: Sounds good, will change

File: gnomad_qc/v4/sample_qc/platform_inference.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+                logger.info("Loading test VDS...")
+                vds = gnomad_v4_testset.vds()
+            else:
+                logger.info("Loading full v4 VDS...")
+                vds = get_gnomad_v4_vds(remove_hard_filtered_samples=False)
+
+            logger.info(
+                "Loading calling intervals: %s with padding of %d...",
+                args.calling_interval_name,
+                args.calling_interval_padding,
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v4/sample_qc/platform_inference.py:51]: yup, thank you, I will change

File: gnomad_qc/v4/sample_qc/platform_inference.py [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+        "--overwrite",
+        help="Overwrite output files (default: False).",
+        action="store_true",
+    )
+    parser.add_argument(
+        "--test",
+        help="Use the v4 test dataset instead of the full dataset.",
+        action="store_true",
+    )
+    parser.add_argument(
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad_qc/v4/sample_qc/platform_inference.py:211]: thank you, still in the previous mindset, will change

**Comment by jkgoodrich** [comment]
Sorry, not positive about this pylint error yet

```
gnomad_qc/v2/annotations/generate_ld_data.py:184:13: E1101: Class 'BlockMatrix' has no '_from_java' member (no-member)
```

**Review by klaricch — APPROVED** [review]
LGTM!

**Pull request merged**